addons/mrp/__init__.py

```diff
@@ -31,6 +31,13 @@ def _create_warehouse_data(env):
 
 def uninstall_hook(env):
     warehouses = env["stock.warehouse"].search([])
+    for warehouse in warehouses:
+        sequence_data = warehouse._get_sequence_values()
+        if sequence_data:
+            domain = ['|', '|', ('prefix', '=', sequence_data.get('pbm_type_id')['prefix']),
+                      ('prefix', '=', sequence_data.get('sam_type_id')['prefix']),
+                      ('prefix', '=', sequence_data.get('manu_type_id')['prefix'])]
```

Comment:
It's cleaner to do a search for picking types with `sequence_id` that use that prefix and/or name from the `get_sequence_values` (i.e. you should also check the sequence name because that's what the constraint in stock.picking.type is actually matching against) and then unlink the sequences from there (i.e. see https://github.com/odoo/odoo/pull/134032/commits/d50a643857265a38688dfaf08c14875813e12409)

Also instead of doing multiple searches, you can link all of the domains together and do 1 search

Reply:
@ticodoo   I updated the code.
Here, we get the sequence name from the `_get_sequence_values()` function, and these sequence names are checked into the `ir.sequence` object and deleted from the `ir.sequence` object.

In exceptional cases, the user will duplicate the sequence record that is linked to the `stock.picking.type` object in the `ir.sequence` object, e.g., "YourCompany Sequence production".  When the user reinstalls the `mrp` module in that case, the `_check_sequence_code()` function will trigger and raise an error message because the duplicate records exist in the `ir.sequence` object. So, I am not directly checking the sequence names from the `stock.picking.type` object using the `sequence_id`. Instead, the `stock.picking.type` object directly compares the sequence name in the `ir.sequence` object and deletes it when the user uninstalls the `mrp` module.

In the `stock.picking.type` object, `_check_sequence_code()` checks the sequence names; if the sequence name is a duplicate, then raises the message.

Reply:
@prye-odoo I prefer that we follow the same logic as the fix for uninstalling/reinstalling stock which I linked in my previous message. If the user decides to duplicate a `stock.picking.type` sequence record and not link it to a stock.picking.type, then it can be manually fixed in their db since as you say, it's a very exceptional case. If it ends up being a big problem, then we can fix the constraint that is causing the issue to only include sequences linked to picking types (the original bugfix could be improved).

Also, it's better to do your original domain where you check that each sequence name is `=` because the `'in'` operator for a string is not the same as `=` (it will check if the string _exists_ within the other string). I believe you can use `from odoo.osv.expression import OR` to combine the domains